### DISCUSSION TITLE
Introduce Doxygen for building doc ?

### DISCUSSION THREAD

**Discussion by vinz6751** [pweingar/FoenixMCP#32] [Ideas]
What about using [Doxygen](https://www.doxygen.nl/index.html) for generating documentation from source code annotation ?
It can export to LaTeX (which can export to PostScript/PDF), though I have never done it.

The "paper" documentation tends to lag behind the code because it's remote from it. By using documentation *in* the code, it's easier to keep things up to date.

Thoughts ?

**Comment by pweingar** [comment]
I've done this sort of thing back in the day with javadoc. It's not a bad process. I fairly familiar with LaTeX and used it for the BASIC816 documentation, but I decided I didn't want to fuss with the table codes to get the manual I wanted and just went with Libre. I've no objection to switching over to this style of documentation though... it's not a bad way to do thing.
